hw3/tasks/task31.py

```diff
@@ -0,0 +1,56 @@
+# .. In previous homework task 4, you wrote a cache function that remembers other function output value.
+# .. Modify it to be a parametrized decorator, so that the following code::
+
+# ..     @cache(times=3)
+# ..     def some_function():
+# ..         pass
+
+# .. Would give out cached value up to `times` number only.
+# .. Example::
+
+# ..     @cache(times=2)
+# ..     def f():
+# ..         return input('? ')   # careful with input() in python2, use raw_input() instead
+
+# ..     >>> f()
+# ..     ? 1
+# ..     '1'
+# ..     >>> f()     # will remember previous value
+# ..     '1'
+# ..     >>> f()     # but use it up to two times only
+# ..     '1'
+# ..     >>> f()
+# ..     ? 2
+# ..     '2'
+
+import timeit
+from typing import Callable
+
+
+def cache(times: int) -> Callable:
+    if times < 0:
+        raise ValueError("times must be non-negative")
+
+    def custom_hash(func: Callable, *args, **kwargs):
```

Comment:
in this way, cache does not make difference between calls with different arguments. like 
cached_func(1,2,3) and cached_func(3,4,5) will return the same result here

Reply:
I thought, that it's needed to cache just function calls until calls with specific arguments